students/stuartkershaw/lesson05/mailroom.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+
+import pathlib
+pth = pathlib.Path('./')
+
+donors = {
+    'Anne Hathaway': {
+        'donations': [
+            4000,
+            6250
+        ],
+    },
+    'Geronimo Jackson': {
+        'donations': [
+            150,
+            400,
+            1250
+        ]
+    },
+    'Dingo Dogson': {
+        'donations': [
+            150,
+            300,
+            100
+        ]
+    },
+    'Leo Kottke': {
+        'donations': [
+            75,
+            125
+        ]
+    },
+    'Flask McCreole': {
+        'donations': [
+            1000
+        ]
+    },
+    'Piglet Norquist': {
+        'donations': [
+            20,
+            15,
+            30
+        ]
+    }
+}
+
+
+def compose_email(donor):
+    message_obj = {
+        'donor_name': donor,
+        'donation': donors[donor]['donations'][-1]
+    }
+    message = 'Dear {donor_name}, thanks so much '\
+              'for your generous donation in the amount of: '\
+              '${donation}.'.format(**message_obj)
+    return message
+
+
+def apply_selection(selection):
+    arg_dict = {
+        '1': get_donor_name,
+        '2': show_donor_table,
+        '3': generate_letters,
+        '4': quit
+    }
+    arg_dict.get(selection)()
+
+
+def get_selection():
+    options = 'Please select 1, 2, 3, or 4:\n'\
+              '1) send a thank you\n'\
+              '2) create a report\n'\
+              '3) send letters to everyone\n'\
+              '4) quit\n'
+    while True:
+        try:
+            selection = input(options)
+            if selection in ['1', '2', '3', '4']:
+                apply_selection(selection)
+        except ValueError:
+            input(options)
```

Comment:
in what case would the ValueError be raised?

if it does happen, the user is not captured, is that intended?

recommendation: 
* use `arg_dict[selection]()` and catch `KeyError` in `apply_selection` function and print a message to user then let the loop continue